src/wbe_db_func.r

```diff
@@ -0,0 +1,577 @@
+
+
+##################################
+# conversion script from long table to wide
+#
+# data model v0.1.1
+#
+library(tidyverse)
+library(readxl)
+library(DBI)
+library(RSQLite)
+library(stringr)
+library(glue)
+library(janitor)
+library(docstring)
+library(hash)
+library(dbplyr)
+library(digest)
+library(magrittr)
+
+WBE_DB_CONNS <- c()
+WBE_DB_CREATE_TABLES_SQL_FN <- file.path("src", "wbe_create_tables.sql")
+
+
+
+##############################
+# call digest_vect for a vectorized version of digest
+digest_vect <- Vectorize(digest)
+
+
+
+dbe_db_default_fn <- function(){
+  file.path("data", "db" ,"WBE.db")
+}
+
+
+wbe_db_conn <- function(drv = RSQLite::SQLite(),
+                        db_fn = dbe_db_default_fn(), 
+                        ...){
+  #' returns a connection to a WBE database
+  #' creates and emply db if needed
+  if(! file.exists(db_fn)){
+    wbe_db_create_db_from_script(db_fn = db_fn )
+  }
+  
+  conn <- dbConnect(drv = drv, db_fn, ...)
+  WBE_DB_CONNS <<- c(WBE_DB_CONNS, conn)
+  return(conn)
+}
+
+
+
+
+wbe_db_Disconnect_all <- function(){
+  #'
+  #' removes all connections to the db that we know of
+  #' 
+  
+  n_con <- length(WBE_DB_CONNS)
+  
+  lapply(WBE_DB_CONNS, function(x){dbDisconnect(x)})
+  WBE_DB_CONNS <<- list()
+  
+  message(glue("disconnected  {n_con} connections"))
+}
+#wbe_db_Disconnect_all()
+
+wbe_db_delete_whole_DB <- function(db_fn = dbe_db_default_fn()){
+  #'
+  #' removes all connections, then deletes the file 
+  #' 
+  wbe_db_Disconnect_all()
+  file.remove(db_fn)
+  message(glue("deleted {db_fn}"))
+}
+#wbe_db_delete_whole_DB
+
+
+wbe_db_create_db_from_script <- function(drv = RSQLite::SQLite(),
+                                         full_fn = WBE_DB_CREATE_TABLES_SQL_FN,
+                                         db_fn = dbe_db_default_fn(), 
+                                         ...
+                                         ){
+  #' creates a wbe_db from a script
+  #' 
+  #' 
+  conn <- dbConnect(drv = drv, db_fn, ...)
+  WBE_DB_CONNS <<- c(WBE_DB_CONNS, conn)
+
+  sql_str <- readChar(con = full_fn, 
+                      nchars = file.info(full_fn)$size)
+  
+  queries1 = str_replace_all(sql_str,'--.*$'," ")  ### remove any commented lines
+  queries2 = paste(queries1, collapse = '\n') ### collapse with new lines
+  queries3 = unlist(str_split(queries2,"(?<=;)")) ### separate individual queries
+  
+  
+  for (i in 1:length(queries3)) {
+    dbExecute(conn, queries3[i])
+  }
+  
+  tbls <- paste0(db_list_tables(conn), collapse = ", ")
+  message(glue("database created at:{db_fn}\nWith tables:{tbls}."))
+}
+
+#wbe_db_create_db_from_script()
+#wbe_db_delete_whole_DB()
+
+
+
+
+
+wbe_db_xlsx_to_dfs <- function(full_fn,
```

Comment:
We haven't talked about data type. I was actually thinking we should stick to multiple csv files as opposed to xlsx for the usual reasons of challenges with binary files and proprietary software. 

Or at least default to csv format.

Reply:
good point, there is nothing wrong with providing multiple methods, and is xlsx is easy to work with out of the box and for lab techs to enter into, and switching to and from a well formated xlsx to multiple csv should be trivial.